tests/JMS/Serializer/Tests/Serializer/DeserializeExceptionTest.php

```diff
@@ -0,0 +1,146 @@
+<?php
+/*
+ * Copyright 2015 Ivan Borzenkov <ivan.borzenkov@gmail.com>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace JMS\Serializer\Tests\Serializer;
+
+use Doctrine\Common\Annotations\AnnotationReader;
+use JMS\Serializer\Context;
+use JMS\Serializer\Exception\RuntimeException;
+use Metadata\MetadataFactory;
+use PhpCollection\Map;
+use Symfony\Component\Translation\IdentityTranslator;
+use Symfony\Component\Translation\MessageSelector;
+
+class DeserializeExceptionTest extends BaseSerializationTest
```

Comment:
actually my idea was to add this tests directly into `BaseSerializationTest` and calling them something as `test(notBoolean|notString...)ThrowsException`